backend/src/main/java/com/redhat/cloud/notifications/routers/sources/Secret.java

```diff
@@ -0,0 +1,46 @@
+package com.redhat.cloud.notifications.routers.sources;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * <p>Defines the model for a Secret on the Sources backend. See {@link SourcesService} for the OpenApi Specs that define
+ * it, and some considerations when trying to store an encrypted secret.</p>
+ * <p>All the fields, except for the {@link Secret#username} and {@link Secret#password} ones are ignored if they are
+ * {@code NULL}, since Sources is very strict with non-existing keys in the payload: it will return a bad request for
+ * those requests.</p>
+ */
+public final class Secret {
```

Comment:
We usually add `@JsonIgnoreProperties(ignoreUnknown = true)` to all classes that are used to deserialize incoming REST data. It prevents deserialization failures if the remote API adds new fields without letting us know. You know the Sources API way better than me so I'll let you decide if we need it or not. 😄

Reply:
It shouldn't happen, but I rather include it and be safe than sorry :+1: